api/app/src/external/fhir/shared/index.ts

```diff
@@ -25,3 +32,23 @@ export function downloadedFromCW(doc: DocumentReference): boolean {
 export function downloadedFromHIEs(doc: DocumentReference): boolean {
   return downloadedFromCW(doc);
 }
+
+// Creates a FHIR data query string based on the specified range.
+// For example, if dateFrom="2022-03-23" & dateTo="2024-01-02", result will look like:
+//  "date=ge2022-03-23&date=le2024-01-02"
+export function isoDateRangeToFHIRDateQuery(dateFrom?: string, dateTo?: string): string {
+  const fhirDateQueryBase = "date=";
+  let fhirDateQuery = `${fhirDateQueryBase}`;
+  if (!dateFrom && !dateTo) return "";
+  if (dateFrom) fhirDateQuery += `${Operator.GREATER_THAN_OR_EQUALS}${dateFrom}`;
+  if (dateTo)
+    fhirDateQuery += `${fhirDateQuery ? `&${fhirDateQueryBase}` : ""}${
+      Operator.LESS_THAN_OR_EQUALS
+    }${dateTo}`;
```

Comment:
If we send `...(undefined, "2023-01-01")` (date is just an example), it will return `date=&date=le2023-01-01`, right?

Is this expected?

Also, when an `if` breaks like, I think its a good practice to have open and close curly brackets - just a suggestion.

Reply:
Fixed on both points